core/jobs/batch_jobs/exploration_migration_jobs.py

```diff
@@ -0,0 +1,503 @@
+    ) -> Sequence[base_models.BaseModel]:
+        """Generates newly updated exploration models.
+
+        Args:
+            exp_model: ExplorationModel. The exploration which
+                should be updated.
+            migrated_exploration: Exploration. The migrated exploration
+                domain object.
+            exploration_changes: sequence(ExplorationChange). The
+                exploration changes to apply.
+            exp_rights_model: ExplorationRightsModel. The
+                additional model.
+
+        Returns:
+            sequence(BaseModel). Sequence of models which should be put into
+            the datastore.
+        """
+        updated_exploration_model = (
+            exp_services.populate_exploration_model_fields(
+                exp_model, migrated_exploration))
+        commit_message = (
+            'Update exploration content schema version to %d.'
+        ) % (exp_domain.Exploration.CURRENT_EXP_SCHEMA_VERSION)
+        change_dicts = [change.to_dict() for change in exploration_changes]
+        with datastore_services.get_ndb_context():
+            models_to_put = updated_exploration_model.compute_models_to_commit(
+                feconf.MIGRATION_BOT_USERNAME,
+                feconf.COMMIT_TYPE_EDIT,
+                commit_message,
+                change_dicts,
+                additional_models={'rights_model': exp_rights_model}
+            ).values()
+        datastore_services.update_timestamps_multi(list(models_to_put))
+        return models_to_put
+
+    @staticmethod
+    def _update_exploration_summary(
+        migrated_exploration: exp_domain.Exploration,
+        exp_summary: exp_domain.ExplorationSummary,
+        exp_summary_model: exp_models.ExpSummaryModel,
+        exp_rights: exp_models.ExplorationRightsModel
+    ) -> exp_models.ExpSummaryModel:
+        """Generates newly updated exploration summary model.
+
+        Args:
+            migrated_exploration: Exploration. The migrated exploration
+                domain object.
+            exp_summary: ExplorationSummary. The exploration
+                summary domain object.
+            exp_summary_model: ExpSummaryModel. The exploration
+                summary model to update.
+            exp_rights: ExplorationRightsModel. The exploration
+                rights model.
+
+        Returns:
+            ExpSummaryModel. The updated exploration summary model to put into
+            the datastore.
+        """
+        if exp_summary_model is not None:
+            old_exp_summary = exp_fetchers.get_exploration_summary_from_model(
+                exp_summary_model)
+            ratings = old_exp_summary.ratings or feconf.get_empty_ratings()
+            scaled_average_rating = (
+                exp_services.get_scaled_average_rating(ratings))
+        else:
+            ratings = feconf.get_empty_ratings()
+            scaled_average_rating = feconf.EMPTY_SCALED_AVERAGE_RATING
```

Comment:
This part is not needed because if the exploration model exist we can presume that the summary will exist (at least in this job).

Reply:
I am guessing that this function is getting null exploration summary model. Will check in detail and update today.